src/ui/mainwindow.cpp

```diff
@@ -1465,17 +1465,9 @@ void MainWindow::StopAfterCurrent() {
 }
 
 void MainWindow::closeEvent(QCloseEvent* event) {
-  bool keep_running(false);
-  if (tray_icon_)
-    keep_running =
-        settings_.value("keeprunning", tray_icon_->IsVisible()).toBool();
-
-  if (keep_running && event->spontaneous()) {
-    event->ignore();
-    SetHiddenInTray(true);
-  } else {
-    Exit();
-  }
+  if (!tray_icon_ || !settings_.value("keeprunning", tray_icon_->IsVisible()).toBool())
```

Comment:
This will now crash on systems without a system tray icon.

Reply:
No it won't, if tray_icon_ is null it short circuits to Exit.

Unless there is some circumstance where tray_icon_ is non-null but tray_icon_->IsVisible is an invalid op? But that would have crashed the previous code.

Reply:
Yes your right, my bad